core/mosaic-webpack-fallback-plugin/index.js

```diff
@@ -109,8 +111,11 @@ class FallbackPlugin {
      * @memberof FallbackPlugin
      */
     getRelativePathname(pathname) {
-        const isSrc = new RegExp(escapeRegex(`${path.sep}src${path.sep}`)).test(pathname);
-        const prefix = isSrc ? 'src' : 'public';
+        const { sourceDirectories } = this.options;
+        const prefix = sourceDirectories
+            .map(directory => path.parse(directory).name)
+            .find(directory => new RegExp(escapeRegex(`${path.sep}${directory}${path.sep}`)).test(pathname))
+            || 'public';
```

Comment:
This logic is hard to understand, please comment on this

Reply:
True, it's hard to understand even for me. This part of the code was written without comments, but from what I understand, it searches for folder name in the root directory and tries to find the same directory in the fallback folder inside node_modules so it can fallback to it, or smth like that.
Now it uses sourceDirectories for that and if it doesn't find in them fallback to the 'public' directory. Dunno how to refactor it more efficiently but it works for now.